src/execution/operator/scan/physical_table_scan.cpp

```diff
@@ -108,7 +108,18 @@ SourceResultType PhysicalTableScan::GetData(ExecutionContext &context, DataChunk
 	if (g_state.in_out_final) {
 		function.in_out_function_final(context, data, chunk);
 	}
-	function.in_out_function(context, data, g_state.input_chunk, chunk);
+	switch (function.in_out_function(context, data, g_state.input_chunk, chunk)) {
+	case OperatorResultType::BLOCKED: {
+		auto guard = g_state.Lock();
+		g_state.BlockSource(guard, input.interrupt_state);
+		return SourceResultType::BLOCKED;
+	}
+	default:
```

Comment:
The expected ones can probably `break`, I'd say the others throw either a NotImplemented or Internal exception

Reply:
Here I am not really sure, since currently the behaviour sort of work as intended, and purpose of this PR is adding handling for a yet untreated cases, where the default must `break` to keep functionality (even if logically not right).

Reply:
Discussed in person, currently `BlockSource` returns FINISHED only when that actually does make sense.